Build fail on Arch latest avr-gcc

Using the latest `avr-gcc (7.1.0-1)` I'm getting the following error when trying to compile:

```
% make sqve                                                                                                                                                                                                                                                                       ~/code/qmk_firmware/keyboards/ergodox master
Making ergodox/ez with keymap sqve

avr-gcc (GCC) 7.1.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	   data	    bss	    dec	    hex	filename
      0	  22752	      0	  22752	   58e0	ergodox_ez_sqve.hex

Compiling: keyboards/ergodox/keymaps/sqve/keymap.c                                                  [OK]
Compiling: keyboards/ergodox/ez/ez.c                                                               keyboards/ergodox/ez/ez.c: In function 'init_mcp23018':
keyboards/ergodox/ez/ez.c:60:24: error: increment of a boolean expression [-Werror=bool-operation]
         i2c_initialized++;
                        ^~
cc1: all warnings being treated as errors
 [ERRORS]
 | 
 | 
 | 
make[1]: *** [tmk_core/rules.mk:361: .build/obj_ergodox_ez_sqve/keyboards/ergodox/ez/ez.o] Error 1
Make finished with errors
make: *** [../../Makefile:508: sqve] Error 1
```

Downgrading to version `6.3.0-1` solves the issue.

```
Linking: .build/ergodox_ez_ariasuni.elf                                                             [ERRORS]                                    
 |                                                                                                                                              
 | .build/obj_ergodox_ez_ariasuni/keyboards/ergodox/ez/ez.o : Dans la fonction « ergodox_blink_all_leds » :                                     
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:32 : référence indéfinie vers « ergodox_led_all_off »                          
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:33 : référence indéfinie vers « ergodox_led_all_set »                          
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:34 : référence indéfinie vers « ergodox_right_led_1_on »                       
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:36 : référence indéfinie vers « ergodox_right_led_2_on »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:38 : référence indéfinie vers « ergodox_right_led_3_on »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:40 : référence indéfinie vers « ergodox_right_led_1_off »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:42 : référence indéfinie vers « ergodox_right_led_2_off »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:44 : référence indéfinie vers « ergodox_right_led_3_off »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/ez.c:47 : référence indéfinie vers « ergodox_led_all_off »
 | .build/obj_ergodox_ez_ariasuni/keyboards/ergodox/ez/keymaps/ariasuni/keymap.o : Dans la fonction « matrix_init_user » :
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:162 : référence indéfinie vers « ergodox_led_all_set »
 | .build/obj_ergodox_ez_ariasuni/keyboards/ergodox/ez/keymaps/ariasuni/keymap.o : Dans la fonction « matrix_scan_user » :
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:170 : référence indéfinie vers « ergodox_right_led_1_on »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:173 : référence indéfinie vers « ergodox_right_led_2_on »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:176 : référence indéfinie vers « ergodox_right_led_3_on »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:171 : référence indéfinie vers « ergodox_right_led_1_off »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:174 : référence indéfinie vers « ergodox_right_led_2_off »
 | /home/ariasuni/projets/qmk_firmware/keyboards/ergodox/ez/keymaps/ariasuni/keymap.c:177 : référence indéfinie vers « ergodox_right_led_3_off »
 | collect2: error: ld returned 1 exit status
```

--- a/keyboards/ergodox/ez/ez.c
+++ b/keyboards/ergodox/ez/ez.c
@@ -1,6 +1,30 @@
 #include "ez.h"
 #include "i2cmaster.h"
 
+
+extern inline void ergodox_board_led_on(void);
+extern inline void ergodox_right_led_1_on(void);
+extern inline void ergodox_right_led_2_on(void);
+extern inline void ergodox_right_led_3_on(void);
+extern inline void ergodox_right_led_on(uint8_t led);
+
+extern inline void ergodox_board_led_off(void);
+extern inline void ergodox_right_led_1_off(void);
+extern inline void ergodox_right_led_2_off(void);
+extern inline void ergodox_right_led_3_off(void);
+extern inline void ergodox_right_led_off(uint8_t led);
+
+extern inline void ergodox_led_all_on(void);
+extern inline void ergodox_led_all_off(void);
+
+extern inline void ergodox_right_led_1_set(uint8_t n);
+extern inline void ergodox_right_led_2_set(uint8_t n);
+extern inline void ergodox_right_led_3_set(uint8_t n);
+extern inline void ergodox_right_led_set(uint8_t led, uint8_t n);
+
+extern inline void ergodox_led_all_set(uint8_t n);
+
+
 bool i2c_initialized = 0;
 uint8_t mcp23018_status = 0x20;
 
@@ -57,7 +81,7 @@ uint8_t init_mcp23018(void) {
     // cli();
     if (i2c_initialized == 0) {
         i2c_init();  // on pins D(1,0)
-        i2c_initialized++;
+        i2c_initialized = true;
         _delay_ms(1000);
     }
 

--- a/keyboards/gh60/gh60.c
+++ b/keyboards/gh60/gh60.c
@@ -1,25 +1,39 @@
 #include "gh60.h"
 
+
+extern inline void gh60_caps_led_on(void);
+extern inline void gh60_poker_leds_on(void);
+extern inline void gh60_fn_led_on(void);
+extern inline void gh60_esc_led_on(void);
+extern inline void gh60_wasd_leds_on(void);
+
+extern inline void gh60_caps_led_off(void);
+extern inline void gh60_poker_leds_off(void);
+extern inline void gh60_fn_led_off(void);
+extern inline void gh60_esc_led_off(void);
+extern inline void gh60_wasd_leds_off(void);
+
+
 void led_set_kb(uint8_t usb_led) {
 	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
-	
+
     if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
 		gh60_caps_led_on();
 		} else {
-		gh60_caps_led_off(); 
+		gh60_caps_led_off();
     }
-	
+
     // if (usb_led & (1<<USB_LED_NUM_LOCK)) {
 		// gh60_esc_led_on();
 		// } else {
-		// gh60_esc_led_off(); 
+		// gh60_esc_led_off();
     // }
-	
+
     // if (usb_led & (1<<USB_LED_SCROLL_LOCK)) {
 		// gh60_fn_led_on();
 		// } else {
-		// gh60_fn_led_off(); 	
+		// gh60_fn_led_off();
     // }
 
-	led_set_user(usb_led);	
+	led_set_user(usb_led);
 }

--- a/keyboards/pegasushoof/pegasushoof.c
+++ b/keyboards/pegasushoof/pegasushoof.c
@@ -17,6 +17,14 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "pegasushoof.h"
 
+
+extern inline void ph_caps_led_on(void);
+extern inline void ph_caps_led_off(void);
+
+extern inline void ph_sclk_led_on(void);
+extern inline void ph_sclk_led_off(void);
+
+
 __attribute__ ((weak))
 void matrix_init_user(void) {
 };

--- a/keyboards/xd60/xd60.c
+++ b/keyboards/xd60/xd60.c
@@ -1,5 +1,13 @@
 #include "xd60.h"
 
+
+extern inline void xd60_caps_led_on(void);
+extern inline void xd60_bl_led_on(void);
+
+extern inline void xd60_caps_led_off(void);
+extern inline void xd60_bl_led_off(void);
+
+
 void led_set_kb(uint8_t usb_led) {
 	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
 

